services/utils/apiRequests.js

```diff
@@ -0,0 +1,32 @@
+import { clientEnv } from "@config/schemas/clientSchema";
+
+export async function sendRequest(requestConfig) {
+  const url = `${clientEnv.NEXT_PUBLIC_BASE_URL}${requestConfig.url}`;
+  try {
+    const res = await fetch(url, {
+      method: requestConfig.method ? requestConfig.method : "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: requestConfig.body ? JSON.stringify(requestConfig.body) : null,
```

Comment:
Within the mapper we can then set the body only if there is requestConfig.body instead of setting to null

Reply:
Great idea 👍